### DISCUSSION TITLE
[TUTORIAL] Building 86Box under Fedora 36

### DISCUSSION THREAD

**Discussion by kuolas** [86Box/86Box#2334] [Show and tell]
I have built 86Box under Fedora 36 and want to share how I did it:

1. Install dependencies

> `sudo dnf install gcc gcc-c++ cmake extra-cmake-modules pkgconf-pkg-config ninja-build freetype-devel SDL2-devel libpng-devel openal-soft-devel rtmidi-devel libFAudio-devel qt5-qtbase-devel qt5-linguist qt5-qtbase-private-devel qt5-qtbase-static libevdev-devel`

2.  Download latest 86Box Git version

>`git clone https://github.com/86Box/86Box.git`

3. Inside the newly created **86Box** directory create a **build** directory

> `cd 86Box`
> `mkdir build`

4. Configure the build environment using CMake

> `cmake -B ./build -S ./ --preset PRESET`

Available configure presets:

  "regular"
  "regularndr"
  "optimized"
  "optimizedndr"
  "debug"
  "debugndr"
  "experimental"
  "experimentalndr"

5. Build

> `cmake --build ./build/`

The executalble binary should be found in **/build/src/86Box**

6. (Optional) Clean the build directory if you plan to rebuild with a different preset

> `cmake --build ./build/ --target clean`

7. Execute

To avoid the message:  `qt.qpa.wayland: Wayland does not support QWindow::requestActivate()` (this is a Qt5 bug)

It shoud be executed using the option **XDG_SESSION_TYPE=x11**

> `XDG_SESSION_TYPE=x11 ./build/src/86Box`

If you have the error: 
```
86Box/src/qt/qt_openglrenderer.hpp:31:10: fatal error: QtOpenGLExtensions/QOpenGLExtensions: No such file or directory
   31 | #include <QtOpenGLExtensions/QOpenGLExtensions>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```
It is because you have not installed **qt5-qtbase-static**

Depending in where you run 86Box is where you should put the ROM files, this is an example execution directly from the development environment: 
```

[user@localhost 86Box]$ XDG_SESSION_TYPE=x11 ./build/src/86Box 
#
# 86Box v3.5 logfile, created 2022/05/08 14:18:08
#
# VM: 86Box
#
# Emulator path: /home/user/Development/Community/86Box/build/src/
# Userfiles path: /home/user/Development/Community/86Box/
# ROM path: /home/user/Development/Community/86Box/roms/
# ROM path: /home/user/Development/Community/86Box/build/src/roms/
# ROM path: /home/user/.local/share/86Box/roms/
# ROM path: /home/user/.local/share/flatpak/exports/share/86Box/roms/
# ROM path: /var/lib/user/exports/share/86Box/roms/
# ROM path: /usr/local/share/86Box/roms/
# ROM path: /usr/share/86Box/roms/
# Configuration file: /home/user/Development/Community/86Box/86box.cfg
#
```

**Comment by OlivierDoriath** [comment]
Thank you. I couldn't build before reading this! 😊

**Comment by rderooy** [comment]
There is also a copr repo available for Fedora.
https://copr.fedorainfracloud.org/coprs/rob72/86Box/

Or if you want to build the rpm yourself (incl. ROMs)
https://github.com/86Box/86Box/blob/master/src/unix/assets/86Box.spec
